mat/matrix.go

```diff
@@ -848,20 +848,57 @@ func normLapack(norm float64, aTrans bool) lapack.MatrixNorm {
 
 // Sum returns the sum of the elements of the matrix.
 func Sum(a Matrix) float64 {
-	// TODO(btracey): Add a fast path for the other supported matrix types.
 
-	r, c := a.Dims()
 	var sum float64
 	aU, _ := untranspose(a)
-	if rma, ok := aU.(RawMatrixer); ok {
+	switch rma := aU.(type) {
+	case RawSymmetricer:
+		rm := rma.RawSymmetric()
+		for i := 0; i < rm.N; i++ {
+			// Diagonals count once while off-diagonals count twice.
+			sum += rm.Data[i*rm.Stride+i]
+			for _, v := range rm.Data[i*rm.Stride+i+1 : i*rm.Stride+rm.N] {
+				sum += 2 * v
```

Comment:
Do a single doubling at the end.

Reply:
now breaks out a single doubling and does the diagonal.  neither approach looks totally clean to me honestly.

Reply:
No, but this has fewer multiplies.